src/webview/js/js.js

```diff
@@ -333,6 +372,23 @@ function someVisibleMessage(top: number, bottom: number): ?Element {
   );
 }
 
+/** Returns some message element which is both visible and read, if any. */
+function someVisibleReadMessage(top: number, bottom: number): ?Element {
+  function checkRead(candidate: ?Element): ?Element {
+    return candidate && isRead(candidate, top, bottom) ? candidate : null;
+  }
+
+  // Algorithm: If there's a visible message that isn't read, that means
+  // it's partway off the bottom of the screen. Therefore, either:
+  // * the message above it will be visible and read, or
+  // * there are no visible read messages.
+  const visible = someVisibleMessage(top, bottom);
+  if (visible === null || visible === undefined) {
+    return visible;
+  }
```

Comment:
nit: can shorten condition to `!visible`

It's important not to do that with a string or number, of course, but in the situation where it's either an object, or null or undefined, then just testing falsiness is equally correct and helps make it not feel like a lot of boilerplate. And the type-checker is there to make sure we know the difference -- in fact it will complain if you use `!foo` when `foo` could be string or number, which is a feature I quite appreciate :slightly_smiling_face:

Reply:
Makes sense :) Writing code like this makes me miss the `?` operator in Rust...

Reply:
Yeah. Kotlin has `?:`, which provides much the same convenience. JS has not done as well at this.